show/platform.py

```diff
@@ -150,11 +151,19 @@ def fan():
 
 # 'temperature' subcommand ("show platform temperature")
 @platform.command()
-def temperature():
+@click.option('-j', '--json', 'output_json', is_flag=True, help="Output in JSON format")
+def temperature(output_json):
     """Show device temperature information"""
     cmd = ['tempershow']
-    clicommon.run_command(cmd)
-
+    if output_json:
+        output, _ = clicommon.run_command(cmd+["-j"], return_cmd=True)
+        try:
+            data = json.loads(output)
+            click.echo(clicommon.json_dump(data))
+        except json.JSONDecodeError:
+            click.echo(output)
```

Comment:
>echo [](http://example.com/codeflow?start=18&length=4)

If no exception or throwing exception, you basically echo the same content. Why not just echo output in all cases?

If you want to treat exception differently, suggest echo to stderr, print some message like "invalid json".